### ISSUE TITLE
global registry entry for non-reTHINK services

### ISSUE THREAD

**Issue by jmcrom** [reTHINK-project/dev-registry-global#20]
could we use GR dataset to store non-reTHINK urls (IMS, slack, skype, etc.) ?

theoretically I do not see any blocking reason, and it will make GR even more universal

see also [here](https://github.com/reTHINK-project/specs/issues/11) and [here](https://github.com/reTHINK-project/dev-discovery/issues/3)

**Comment by sgoendoer** [comment]
Something like this could be very easily added to the dataset. It wouldn't interfere with any interference. Question would be what kinds of legacy services we would want to be allowed to be added...

```
{
    //...
    "legacy": [
        {"type": "phone", "id": "+493012345678"},
        {"type": "facebook", "id": "http://facbook.com/fluffy"}
    ]
}
```

**Comment by jmcrom** [comment]
to the extreme: any means of contact, including:
- mail
- phone/sip/ims
- OSN wall pages: facebook/google+/linkedin

**Comment by sbecot** [comment]
It may be difficult to be comprehensive, so a type for an "anykind" of service could be reserved.
{"type": "unknown", "id": "john.doe", "url":"http://mywonderfullapplication/"}

For facebook it would be more like
        {"type": "facebook", "id": "fluffy", "url":"http://facebook.com/"}
or 
        {"type": "facebook", "id": "fluffy"}

**Comment by sgoendoer** [comment]
Before we finally implement this feature, I'd like to discuss and fix a data format here. So here goes a first proposal from my side:

### intention

The intention behind this feature is to allow people to discover different /legacy ways of contacting a person.

### use case

Alice looks up Bob's GUID. As he has no running reThink-compatible endpoint running at the moment, she tries calling him on his landline number, which he provided as a legacy contact information in the dataset.

### requirements

- datasets should provide no, one, or many alternative ways to contact a user outside of the rethink ecosystem.
- legacy ids can by of any type
- legacy ids should have 
> - a type that describes how to "use" the identifier (e.g. "email" or "phone")
> - a category that makes identifiers distinguishable (e.g. "work phone" or "private cellphone")
> - human readable description (e.g. "work phone number at Evil Corp")
> - the identifier itself (e.g. "+123987654321" or "someone@googlemail.com")

### proposed example

```
"legacy": [{
    "type": "email",
    "category": "work",
    "description": "please dont spam me too much",
    "id": "someone@googlemail.com"
},{
    "type": "phone",
    "category": "private",
    "description": "i wont pick up anyways",
    "id": "+123456789123"
}]
```

Any more ideas?

**Comment by sgoendoer** [comment]
We probably need to specify further what "type" can be

Type: 
- website / url
- email
- phone

do we want to allow "offline" identifiers too? Like

- postal address
- coordinates (lat-lng)

**Comment by sgoendoer** [comment]
This has been implemented in the upcoming version 0.3.x
